test/test_datasets.py

```diff
@@ -141,76 +145,68 @@ def test_places365_repr_smoke(self):
 
-        def target_transform(target):
-            return expected_target
+        def make_label_file(num_images, root, name):
+            make_binary_file(num_images, root, name)
 
-        with self.mocked_dataset(transform=transform, target_transform=target_transform) as (dataset, _):
-            actual_image, actual_target = dataset[0]
+        def make_class_names_file(root, name="class_names.txt"):
+            class_names = ("airplane", "bird")
+            with open(os.path.join(root, name), "w") as fh:
+                for cname in class_names:
+                    fh.write(f"{cname}\n")
 
-            self.assertEqual(actual_image, expected_image)
-            self.assertEqual(actual_target, expected_target)
+        def make_fold_indices_file(root):
+            num_folds = 10
+            offset = 0
+            with open(os.path.join(root, "fold_indices.txt"), "w") as fh:
+                for fold in range(num_folds):
+                    line = " ".join([str(idx) for idx in range(offset, offset + fold + 1)])
+                    fh.write(f"{line}\n")
+                    offset += fold + 1
 
-    def test_unlabeled(self):
-        with self.mocked_dataset(split="unlabeled") as (dataset, _):
-            labels = [dataset[idx][1] for idx in range(len(dataset))]
-            self.assertTrue(all([label == -1 for label in labels]))
+            return tuple(range(1, num_folds + 1))
 
-    @unittest.mock.patch("torchvision.datasets.stl10.download_and_extract_archive")
-    def test_download_preexisting(self, mock):
-        with self.mocked_dataset(pre_extract=True) as (dataset, data):
-            mock.assert_not_called()
+        def make_train_files(root, num_unlabeled_images=1):
+            num_images_in_fold = make_fold_indices_file(root)
+            num_train_images = sum(num_images_in_fold)
 
-    def test_repr_smoke(self):
-        with self.mocked_dataset() as (dataset, _):
-            self.assertIsInstance(repr(dataset), str)
+            make_image_file(num_train_images, root, "train_X.bin")
+            make_label_file(num_train_images, root, "train_y.bin")
+            make_image_file(1, root, "unlabeled_X.bin")
+
+            return dict(train=num_train_images, unlabeled=num_unlabeled_images)
+
+        def make_test_files(root, num_images=2):
+            make_image_file(num_images, root, "test_X.bin")
+            make_label_file(num_images, root, "test_y.bin")
+
+            return dict(test=num_images)
+
+        def make_archive(root, name):
+            make_tar(root, name, name, compression="gz")
+
+        archive_name = "stl10_binary"
+        archive_folder = os.path.join(tmpdir, archive_name)
+        os.mkdir(archive_folder)
+
+        num_images_in_split = make_train_files(archive_folder)
+        num_images_in_split.update(make_test_files(archive_folder))
+        make_class_names_file(archive_folder)
+        make_archive(tmpdir, archive_name)
```

Comment:
Following the docs for `inject_fake_data`

https://github.com/pytorch/vision/blob/eee424922e998dddcaabe927e823ec515aa2927b/test/datasets_utils.py#L308-L310

we should not create an archive here, but rather provide the data in the format after the extraction.